apps/open-swe-v2/src/post-model-hook.ts

```diff
@@ -0,0 +1,100 @@
+import { AIMessage, AIMessageChunk } from "@langchain/core/messages";
+import { interrupt } from "@langchain/langgraph";
+import { WRITE_COMMANDS } from "./constants.js";
+import { AgentStateHelpers, type CodingAgentStateType } from "./state.js";
+import { ToolCall } from "@langchain/core/messages/tool";
+import { ApprovedOperations } from "./types.js";
+
+type StateType = CodingAgentStateType;
+
+export function createAgentPostModelHook() {
+  /**
+   * Post model hook that checks for write tool calls and uses caching to avoid
+   * redundant approval prompts for the same command/directory combinations.
+   */
+  async function postModelHook(state: StateType): Promise<StateType> {
```

Comment:
ah the full vs partial issue might be here. node return types should always be:
```suggestion
  async function postModelHook(state: StateType): Promise<Partial<StateType>> {
```